crates/sncast/tests/e2e/declare.rs

```diff
@@ -266,6 +267,47 @@ async fn test_contract_already_declared() {
     );
 }
 
+#[tokio::test]
+async fn test_contract_already_declared_estimate_fee() {
+    let contract_path = duplicate_contract_directory_with_salt(
+        CONTRACTS_DIR.to_string() + "/map",
+        "put",
+        "74362345",
+    );
+    let tempdir = create_and_deploy_oz_account().await;
+    join_tempdirs(&contract_path, &tempdir);
+
+    let args = vec![
+        "--accounts-file",
+        "accounts.json",
+        "--account",
+        "my_account",
+        "--wait",
+        "declare",
+        "--url",
+        URL,
+        "--contract-name",
+        "Map",
+    ];
+    // Commented out because we explicitly do not want to set any resource bounds.
+    // Transaction has to go through estimate fee endpoint first because it throws different errors
+    // than the normal declare endpoint and we want to test them.
+    // let args = apply_test_resource_bounds_flags(args);
```

Comment:
```suggestion
    // We explicitly do not want to set any resource bounds.
    // Transaction has to go through estimate fee endpoint first because it throws different errors
    // than the normal declare endpoint and we want to test them.
```